app/javascript/schools/courses/types/curriculum_editor/TargetChecklistItem.re

```diff
@@ -0,0 +1,196 @@
+type choices = array(string);
+
+type kind =
+  | Files
+  | Link
+  | ShortText
+  | LongText
+  | MultiChoice(choices);
+
+type t = {
+  title: string,
+  kind,
+  optional: bool,
+};
+
+let title = t => t.title;
+let kind = t => t.kind;
+let optional = t => t.optional;
+
+let actionStringForKind = kind => {
+  switch (kind) {
+  | Files => "Upload Files"
+  | Link => "Attach a Link"
+  | ShortText => "Write Short Text"
+  | LongText => "Write Long Text"
+  | MultiChoice(_choices) => "Choose from a list"
+  };
+};
+
+let kindAsString = kind => {
+  switch (kind) {
+  | Files => "files"
+  | Link => "link"
+  | ShortText => "shortText"
+  | LongText => "longText"
+  | MultiChoice(_choices) => "multiChoice"
+  };
+};
+
+let make = (~title, ~kind, ~optional) => {
+  {title, kind, optional};
+};
+
+let updateTitle = (title, t) => {
+  {...t, title};
+};
+
+let updateKind = (kind, t) => {
+  {...t, kind};
+};
+
+let updateOptional = (optional, t) => {
+  {...t, optional};
+};
+
+let removeItem = (index, list) => {
+  list |> Js.Array.filteri((_item, i) => i != index);
+};
+
+let moveUp = (t, list) => {
+  list |> Array.to_list |> ListUtils.swapUp(t) |> Array.of_list;
+};
+
+let moveDown = (t, list) => {
+  list |> Array.to_list |> ListUtils.swapDown(t) |> Array.of_list;
+};
+
+let copy = (t, list) => {
+  list
+  |> Array.map(item => item == t ? [item, item] : [item])
+  |> ArrayUtils.flatten;
+};
+
+let removeMultichoiceOption = (choiceIndex, t) => {
+  switch (t.kind) {
+  | MultiChoice(choices) =>
+    let updatedChoices =
+      choices
+      |> Array.mapi((i, choice) => i == choiceIndex ? [] : [choice])
+      |> ArrayUtils.flatten;
+    t |> updateKind(MultiChoice(updatedChoices));
+  | Files
+  | Link
+  | ShortText
+  | LongText => t
+  };
+};
+
+let addMultichoiceOption = t => {
+  switch (t.kind) {
+  | MultiChoice(choices) =>
+    let updatedChoices = [|""|] |> Array.append(choices);
+    t |> updateKind(MultiChoice(updatedChoices));
+  | Files
+  | Link
+  | ShortText
+  | LongText => t
+  };
+};
+
+let updateMultichoiceOption = (choiceIndex, newOption, t) => {
+  switch (t.kind) {
+  | MultiChoice(choices) =>
+    let updatedChoices =
+      choices
+      |> Array.mapi((i, choice) => i == choiceIndex ? newOption : choice);
+    t |> updateKind(MultiChoice(updatedChoices));
+  | Files
+  | Link
+  | ShortText
+  | LongText => t
+  };
+};
+
+let createNew = {title: "", kind: LongText, optional: false};
```

Comment:
I would probably just call this binding `longText`. The `createNew` seems to indicate some sort of action - this isn't even a function.